jbpm-flow/src/main/java/org/jbpm/workflow/instance/node/RuleSetNodeInstance.java

```diff
@@ -72,32 +77,61 @@ public void internalTrigger(final NodeInstance from, String type) {
     	if ( !org.jbpm.workflow.core.Node.CONNECTION_DEFAULT_TYPE.equals( type ) ) {
             throw new IllegalArgumentException( "A RuleSetNode only accepts default incoming connections!" );
         }
-    	// first set rule flow group
-    	setRuleFlowGroup(resolveRuleFlowGroup(getRuleSetNode().getRuleFlowGroup()));
-    	
-    	//proceed
+    	RuleSetNode ruleSetNode = getRuleSetNode();
     	KnowledgeRuntime kruntime = getProcessInstance().getKnowledgeRuntime();
-    	Map<String, Object> inputs = evaluateParameters(getRuleSetNode());
-    	for (Entry<String, Object> entry : inputs.entrySet()) {
-    	    String inputKey = getRuleFlowGroup() + "_" +getProcessInstance().getId() +"_"+entry.getKey();
-    	    
-    	    factHandles.put(inputKey, kruntime.insert(entry.getValue()));
-    	}
+    	Map<String, Object> inputs = evaluateParameters(ruleSetNode);
     	
-    	if (actAsWaitState()) {
-        	addRuleSetListener();
-            ((InternalAgenda) getProcessInstance().getKnowledgeRuntime().getAgenda())
-            	.activateRuleFlowGroup( getRuleFlowGroup(), getProcessInstance().getId(), getUniqueId() );
-            
+    	if (ruleSetNode.isDMN()) {
+    	    String namespace = resolveVariable(ruleSetNode.getNamespace());
+    	    String model = resolveVariable(ruleSetNode.getModel());
+    	    String decision = resolveVariable(ruleSetNode.getDecision());
+    	    
+    	    DMNRuntime runtime = ((StatefulKnowledgeSession) kruntime).getKieRuntime(DMNRuntime.class);
+    	    DMNModel dmnModel = runtime.getModel(namespace, model);
+    	    if (dmnModel == null) {
+    	        throw new IllegalArgumentException( "DMN model '" + model + "' not found with namespace '" + namespace + "'");
+    	    }
```

Comment:
Good point, we might need to refine the code here:

https://github.com/droolsjbpm/kie-dmn/blob/master/kie-dmn-core/src/main/java/org/kie/dmn/core/assembler/DMNAssemblerService.java#L55

If the model completely fails to compile and returns null, then it will add an error to the kbuilder, but if it does return a result that contains errors (the most common case) it will not add the errors to the kbuilder.

We might need to do some testing to evaluate the best way to handle this case.